### PR TITLE
Allow users to provide extra rows in the top table

### PR DESCRIPTION
**Pull request by gusgordon**
This allows users to pass in extra rows (as a `dict` or `OrderedDict`) to display in the perf_stats table. They can pass `header_rows` to `create_full_tear_sheet`, `create_simple_tear_sheet`, `create_returns_tear_sheet`, `show_perf_stats`, and `print_table`. HTML will be generated and injected into the top of the table. Also puts the dates/ranges here, below the rows that the user passes.

Also clean up a few random doc strings.

Examples:
screen shot 2017-09-27 at 1 14 14 pm

### CODE REVIEW

File: pyfolio/tears.py [hunk 1]
```diff
@@ -81,7 +81,7 @@ def create_full_tear_sheet(returns,
                            factor_returns=None,
                            factor_loadings=None,
                            pos_in_dollars=True,
-                           return_fig=False):
```
Comment (twiecki) [inline pyfolio/tears.py]: is that right?

**Comment by twiecki** [comment]
Great feature. Is this off of recent master?

File: pyfolio/tears.py (hunk 1 above)
Comment (gusgordon) [reply pyfolio/tears.py]: I think so, for some reason we were accepting this argument but it wasn't being used anymore. return_fig is only used for the create_tear_sheet functions that produce one figure with multiple plots, like the returns tear sheet (the full tear sheet produces a bunch of figures). I am surprised this got by flake8 in the past.

**Comment by gusgordon** [comment]
Yeah off recent master. I will re-run tests once Travis starts passing on master, I think @vikram-narayan is working on that.

**Comment by joshpayne** [comment]
Cool! I'm curious - what problem is this solving for the pyfolio user?

**Comment by gusgordon** [comment]
@joshpayne we can use this internally to print metadata (code name, revision, etc.) about the algos within the tear sheet. Users can print their name, etc in here if they want to show this tear sheet to people. It also formats the dates at the top using the same function, whereas before they were just strings tacked on.

**Pull request merged**